h/services/search_index/_queue.py

```diff
@@ -71,6 +74,14 @@ def sync(self, limit):
         if not jobs:
             return
 
+        # Annotation IDs are stored in the DB's internal hex format in the job
+        # table but we need them in the URL-safe format used by
+        # models.Annotation and in Elasticsearch.
+        for job in jobs:
+            job.kwargs["annotation_id"] = URLSafeUUID().process_result_value(
+                job.kwargs["annotation_id"], "dialect"
+            )
```

Comment:
This modifies `job.kwargs["annotation_id"]` in Python but the modification won't be saved back to the DB because of the lack of mutation tracking for JSON columns. In this case that's what we actually want.

`"dialect"` is unused.

Might want to refactor the code in `types.py` so I don't have to do `URLSafeUUID().process_result_value(value, "dialect")`. Provide a `to_urlsafe(value)` function